### ISSUE TITLE
RPMv6 proposal: require tag data entries in package headers (both main and signature) to be sorted

### ISSUE THREAD

**Issue by DemiMarie** [rpm-software-management/rpm#1555]
Currently, tag data entries in package headers are not required to be sorted.  However, RPM has always produced them in sorted order, and requiring makes detection of duplicate entries trivial.

**Comment by pmatilai** [comment]
Yeah - always except when buggy. There have been at least two occasions in the past where this was broken, which is why rpm currently has to accept them. v6 would be a good point to break and enforce this from the start, thanks for the suggestion.

**Comment by DemiMarie** [comment]
How will this interact with signature headers that are not in a contiguous region?  My personal preference is to require signature headers to be contiguous in RPMv6, and not doing so seems to interact poorly with this proposal.  That said, you did NAK my earlier suggestion to that effect.
